include/triton/core/tritonbackend.h

```diff
@@ -372,6 +372,14 @@ TRITONBACKEND_DECLSPEC TRITONSERVER_Error* TRITONBACKEND_OutputBufferAttributes(
 TRITONBACKEND_DECLSPEC TRITONSERVER_Error* TRITONBACKEND_RequestId(
     TRITONBACKEND_Request* request, const char** id);
 
+/// Get the ID of the request as a string formatted for logging.
+///
+/// \param request The inference request.
+/// \param formatted_id Returns the formatted string of the ID.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONBACKEND_DECLSPEC TRITONSERVER_Error* TRITONBACKEND_RequestIdString(
+    TRITONBACKEND_Request* request, const char** formatted_id);
```

Comment:
Why not reusing `TRITONBACKEND_RequestId` above?

Reply:
I would like to think about it more before adding new public-facing API. For now, we can provide common utility to format the request ID returned from existing APIs.

Reply:
What do you mean by providing a common utility? Repositories other than core seem to use TRITONBACKEND_Request or TRITONSERVER_Request objects rather than InferenceRequest objects directly.

I'm happy to use an alternative approach. It seems like the two main options would be:

1. Create a function in the common repo just for formatting the request ID.
2. Replicate the formatting function in each repository we want to use it in (backend, server, and any backends we decide to add logging to). Right now, the formatting is happening in one place, so any changes to how we print request IDs or similar can be done by changing IdString() in InferenceRequest. This could could create redundancy and different formatting over time.

I think the current change avoids issues with those two approaches, but I get the desire to avoid updating the API for a logging ticket. Take your time on thinking about it, I can change the approach as needed.

Reply:
I think it would be better to have an internal helper (can be in `common` repo if the helper is used across multiple repos server/core/backend/etc.) for formatting the request string if needed, rather than exposing another public C API method we would need to then maintain for backward compatibility and such, as Guan suggests.

Users that want to do this kind of logging with C API in custom backend themselves can just make use of the existing `TRITONBACKEND_RequestId` to get the ID when logging. We don't need to provide that API for them.

Reply:
Thank you both for your comments and insights. I removed the API changes. Upon some more thought, we're primarily using this in `core` (where the API calls are unnecessary) and `backends` (where we have a common utility) for now. Hopefully, this and the related PRs look cleaner now.